platform/mbed_wait_api.h

```diff
@@ -53,18 +53,27 @@ extern "C" {
  *  the accuracy of single precision floating point).
  *
  *  @param s number of seconds to wait
+ * 
+ *  @note
+ *    This func will lock the deep sleep, if you want device entry deep sleep mode, please use Thread::wait()
```

Comment:
Minor grammer nits.

"This func will lock deep sleep. If you want device entry deep sleep mode, please use Thread::wait()"